controllers/redis/redis_controller.go

```diff
@@ -172,6 +180,36 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 		return rbacResult, nil
 	}
 
+	// Redis config maps
+	configMapVars := make(map[string]env.Setter)
+	err = r.generateConfigMaps(ctx, helper, instance, &configMapVars)
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.ServiceConfigReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.ServiceConfigReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, fmt.Errorf("error calculating configmap hash: %w", err)
+	}
+	instance.Status.Conditions.MarkTrue(condition.ServiceConfigReadyCondition, condition.ServiceConfigReadyMessage)
+
+	// the headless service provides DNS entries for pods
+	// the name of the resource must match the name of the app selector
+	pkghl := redis.HeadlessService(instance)
```

Comment:
Ack the reason I didn't do that for the headless service was that the lib-common's counterpart doesn't expose the PublishNotReadyAddresses for creating DNS entries for pods even when their healthcheck return not ready, which we need for sentinel failover to work as expected.
 
I just proposed https://github.com/openstack-k8s-operators/lib-common/pull/444 to add the missing bits. This could be used as well for the galera controller.

Reply:
ack thats merged now

Reply:
as discussed can do that in the followup